streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -513,24 +532,33 @@ private TopicPartition getTopicPartition(final String topicName) {
         return topicPartition;
     }
 
-    private void captureOutputRecords() {
+    private boolean captureOutputRecordsAndEnqueueInternalForwards() {
```

Comment:
This fix seems rather complex. Why not just loop over `output` twice? In the first loop, we only remove output records that don't need further processing, and in the second loop, we call `pipeRecord()` directly for each?

Reply:
Thanks for the review. At a glance, that does seem like it would result in the right output for this particular topology, but it seems like this would still process intermediate results in LIFO order instead of FIFO.

I agree that the current fix is a bit complex, mostly because we can't actually query whether the task's priority queue is empty. I can explore adding that capability, which would simplify this algorithm.

I'll give that a shot after I factor out the TTD change as a separate PR.